easybuild/tools/modules.py

```diff
@@ -742,14 +742,44 @@ def modpath_extensions_for(self, mod_names):
             r'^\s*module\s+use\s+"?([^"\s]+)"?',  # 'module use' in Tcl module files
             r'^\s*prepend-path\s+MODULEPATH\s+"?([^"\s]+)"?',  # prepend to $MODULEPATH in Tcl modules
             r'^\s*prepend_path\(\"MODULEPATH\",\s*\"(\S+)\"',  # prepend to $MODULEPATH in Lua modules
+            r'^\s*prepend_path\(\"MODULEPATH\",\s*pathJoin\((.+)\)\)',  # prepend to $MODULEPATH in Lua modules using pathJoin
         ])
         modpath_ext_regex = re.compile(modpath_ext_regex, re.M)
 
         modpath_exts = {}
         for mod_name in mod_names:
             modtxt = self.read_module_file(mod_name)
-            exts = [ext for tup in modpath_ext_regex.findall(modtxt) for ext in tup if ext]
+            matches = modpath_ext_regex.findall(modtxt)
+            exts = [ext for tup in matches for ext in tup if ext]
             self.log.debug("Found $MODULEPATH extensions for %s: %s", mod_name, exts)
+
+            # Need to expand environment variables and join paths, e.g. when --subdir-user-modules is used
+            for i, ext in enumerate(exts):
+
+                if not matches[i][2]: # can skip plain literal Lua match
```

Comment:
It works here because the matches are mutually exclusive. So the matches list looks like this:
[('','','','match'),('match','','',''),...]
only one of the 4 is filled in.